sympy/physics/vector/frame.py

```diff
@@ -554,45 +522,182 @@ def orient(self, parent, rot_type, amounts, rot_order=''):
+		"""Sets the orientation of this reference frame relative to another
+        (parent) reference frame by setting the direction cosine matrix directly.
+        
+        Parameters
+        ==========
+        
+        parent : ReferenceFrame
+        	Reference frame that this referece frame will be rotated relative to.
+        	
+    	amounts : Matrix, shape(3, 3)
+    	
+    	Examples
+    	==========
+    	
+    	Setup variables for the examples:
 
-        >>> B.orient(N, 'Axis', (q1, N.x))
+        >>> from sympy import symbols
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> q1 = symbols('q1')
+        >>> B = ReferenceFrame('B')
+        >>> N = ReferenceFrame('N')
+        
+    	The orientation of frame A can be set to be the same as frame B in the following way:
+    	
+    	>>> B.orient_axis(N, (q1, N.x))
         >>> A = ReferenceFrame('A')
-        >>> A.orient(N, 'DCM', N.dcm(B))
+        >>> A.orient_dcm(N, N.dcm(B))
         >>> A.dcm(N)
         Matrix([
         [1,       0,      0],
         [0,  cos(q1), sin(q1)],
         [0, -sin(q1), cos(q1)]])
 
-        **Note carefully that** ``N.dcm(B)`` **was passed into** ``orient()``
+        **Note carefully that** ``N.dcm(B)`` **was passed into** ``orient_dcm()``
         **for** ``A.dcm(N)`` **to match** ``B.dcm(N)``.
+    	"""
+        from sympy.physics.vector.functions import dynamicsymbols
+        _check_frame(parent)
+        # amounts must be a Matrix type object
+        # (e.g. sympy.matrices.dense.MutableDenseMatrix).
+        if not isinstance(amounts, MatrixBase):
+            raise TypeError("Amounts must be a sympy Matrix type object.")
+            
+	    parent_orient_dcm = []
+    	parent_orient_dcm = amounts
 
-        Body
-        ----
+        # Reset the _dcm_cache of this frame, and remove it from the
+        # _dcm_caches of the frames it is linked to. Also remove it from the
+        # _dcm_dict of its parent
+        frames = self._dcm_cache.keys()
+        dcm_dict_del = []
+        dcm_cache_del = []
+        for frame in frames:
+            if frame in self._dcm_dict:
+                dcm_dict_del += [frame]
+            dcm_cache_del += [frame]
+        for frame in dcm_dict_del:
+            del frame._dcm_dict[self]
+        for frame in dcm_cache_del:
+            del frame._dcm_cache[self]
+        # Add the dcm relationship to _dcm_dict
+        self._dcm_dict = self._dlist[0] = {}
+        self._dcm_dict.update({parent: parent_orient_dcm.T})
+        parent._dcm_dict.update({self: parent_orient_dcm})
+        # Also update the dcm cache after resetting it
+        self._dcm_cache = {}
+        self._dcm_cache.update({parent: parent_orient_dcm.T})
+        parent._dcm_cache.update({self: parent_orient_dcm})
+        wvec = self._w_diff_dcm(parent)
+        self._ang_vel_dict.update({parent: wvec})
+        parent._ang_vel_dict.update({self: -wvec})
+        self._var_dict = {}
```

Comment:
You seem to have a lot of repeated code in these methods. The repeated code should be in a single hidden method so there is no redundancy.

Reply:
Thank you for that, I would certainly try making a new method to prevent the repetition of code.